examples/map.go

```diff
@@ -0,0 +1,40 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/AlecAivazis/survey"
+)
+
+// the questions to ask
+var simpleQs = []*survey.Question{
+	{
+		Name: "name",
+		Prompt: &survey.Input{
+			Message: "What is your name?",
+		},
+		Validate: survey.Required,
+	},
+	{
+		Name: "color",
+		Prompt: &survey.Select{
+			Message: "Choose a color:",
+			Options: []string{"red", "blue", "green"},
+		},
+		Validate: survey.Required,
+	},
+}
+
+func main() {
```

Comment:
So i've been thinking about this more and more and I'm not convinced that this is a good api to add to `survey` since it tends to a lot of type assertions. Do you mind adding an example that cannot be accomplished with a `struct`? My gut says that if the want is for a dynamic set of questions, that can be accomplished with #5. I'm definitely interested in exploring this, just want to make sure we make very deligent additions to the API, removing them is hard.